math/singular/Portfile

```diff
@@ -4,10 +4,10 @@ PortSystem          1.0
 
 name                singular
 version             4.2.1
-revision            0
+revision            1
 categories          math
 platforms           darwin
-maintainers         nomaintainer
+maintainers         {@catap korins.ky:kirill} openmaintainernomaintainer
```

Comment:
`openmaintainernomaintainer` doesn't look right ;) and increasing the `revision` is unnecessary

Reply:
`openmaintainernomaintainer` fixed. But this port is depends on `flint` which I've upgraded on the same commit. Are you sure that increasing revision to rebuild everything isn't necessary here?

Reply:
that depends: did the library version change and/or is it ABI incompatible? If so, then yes otherwise probably not. You can either look at the installed files, or if you have the port already installed and do the update of flint, then at the end the "rev-upgrade" part will/should warn about broken ports and will rebuild them. That would surely be an indication that the revbump is needed. 

So in short: I don't know for sure and you'd need to figure out if it's needed. Doing a revbump "just in case" is not really ideal.

Reply:
@reneeotten yes, it is my point. Both `flint` and `ntl` is changed a bit API and ABI, and `singular` depends on both of them.

If I install `singular`, after that upgrade both libraries, and `port rev-upgrade` says that `singular` should be rebuild.

I guess this is the case when `revision` should be increased.

Reply:
okay, so then that's correct. But you don't mention the reason for the revbump, but only say that you fixed the livecheck and added yourself as maintainer. 

So in this case three commits, one for each port, would be best and then you can add in the commit message the revbump because of library change. Then I/we don't have to guess 😉

Reply:
@reneeotten like this?

Reply:
not completely, the description is now clear though. But ideally you should have a commit for every port your changing (and with manor changes it's sometimes better to have multiple commits even if you change a single port, but that's irrelevant here). 

So you should separate the changes here so that each commit touches only a single port (i.e., three commits in this PR).

Reply:
@reneeotten but if I touch library all depended should be rebuilded. Idea of having 3 commits was simple: keep it atomic and keep everything as single commit.

Reply:
sure, you want to keep them in the same PR but it should be in different commits

Reply:
@reneeotten it is clear.